tests/test_agent_manager.py

```diff
@@ -66,6 +70,7 @@ def __init__(self, sid, tid, endpoint_names: Set[str], nodename):
         self.nodename = nodename
         self.client = Mock()
         self.client.set_state.side_effect = empty_future
+        self.client.get_status = api_call_future
```

Comment:
Have you tried `self.client.get_status = AsyncMock(return_value=Result(200, "X"))`? If I understand correctly this should have the same result but it would be more clear that we set this method for mocking purposes. Also, there would be no need to define the method above, which would be a plus because the method's purpose is not immediately clear.
If this does not work, perhaps add a short docstring to the method above explaining this is a mock method.

Reply:
AsyncMock doesn't exist for Python 3.6, I will add a docstring.

Reply:
Right, you mentioned that before, I forgot.

Reply:
Did it in https://github.com/inmanta/inmanta/pull/2521/commits/8115141e942dc11443a6149c5b79a845a68f08d5